File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
       if (nums.length === 0) return 0;

        let numsFinal = Array.from(new Set(nums.sort((a, b) => a - b)));

        let maxCount = 1;
        let count = 1;

        for (let i = 1; i < numsFinal.length; i++) {

            if (numsFinal[i] === numsFinal[i - 1] + 1) {
                count++;
            } else {
                count = 1;
            }

            maxCount = Math.max(maxCount, count);
        }

        return maxCount;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int i = 1;
        int j = 0;

        for (int p : piles) {
            j = Math.max(j, p);
        }

        while (i <= j) {
            int mid = i + (j - i) / 2;
            int time = 0;

            for (int x = 0; x < piles.length; x++) {
                time += (piles[x] + mid - 1) / mid;
            }

            if (time > h) {
                i = mid + 1;
            } else {
                j = mid - 1;
            }
        }

        return i;
    }
}